src/tests/Register.test.js

```diff
@@ -1,202 +1,321 @@
-    fireEvent.click(screen.getByLabelText('mentee', { name: "need_mentoring" }), {
-        target: { value: true },
-    })
-    expect(screen.getByLabelText('termsCheck', { name: 'terms_and_conditions_checked'}).checked).toEqual(false)
-    fireEvent.click(screen.getByLabelText('termsCheck', { name: "terms_conditions_checked" }), {
-        target: { value: true },
-    })
-    
-    fireEvent.click(screen.getByRole('button', { name: "Sign Up" }), {
-        target: { value: 'true' },
-    })
-
-    expect(screen.queryByLabelText("response")).not.toBeInTheDocument();
-    
-})
-
-it('handles password toggle', () => {
-    render(<Router><Register /></Router>)
-
-    expect(screen.getByPlaceholderText('Password').type).toEqual("password")
-    expect(screen.getByPlaceholderText('Confirm Password').type).toEqual("password")
-
-    fireEvent.click(screen.getByLabelText('Show Password', { name: "show_password_checkbox" }), {
-        target: { value: true },
-    })
-
-    expect(screen.getByPlaceholderText('Password').type).toEqual("text")
-    expect(screen.getByPlaceholderText('Confirm Password').type).toEqual("text")
-})
-
-it('checks empty field warning', async () => {
-   render(<Router><Register /></Router>)
-
-   fireEvent.change(screen.getByLabelText("Name :", { selector: "input" }), {
-        target: { value: '' },
-    })
-
-   fireEvent.change(screen.getByLabelText("Username :", { selector: "input" }), {
-        target: { value: '' },
-    })
-
-   fireEvent.change(screen.getByLabelText("Email :", { selector: "input" }), {
-        target: { value: '' },
-    })
-
-   fireEvent.change(screen.getByLabelText("Password :", { selector: "input" }), {
-        target: { value: '' },
-    })
-    
-   fireEvent.change(screen.getByLabelText("Confirm Password :", { selector: "input" }), {
-        target: { value: '' },
-    })
-
-   act(() => {
-        fireEvent.click(screen.getByRole('button', { name: "Sign Up" }), {
-            target: { value: 'true' },
-        })
+  rest.post(`${BASE_API}/register`, (req, res, ctx) => {
+    req.body = {
+      fname: "My Name",
+      username: "MyUsername",
+      password: "12345678",
+      email: "myemail@xxx.xxx",
+      terms_and_conditions_checked: true,
+      need_mentoring: true,
+      available_to_mentor: true,
+    };
+
+    return res(
+      ctx.json({
+        message:
+          "User was created successfully. A confirmation email has been sent via email. After confirming your email you can login.",
+      })
+    );
+  })
+);
+
+beforeAll(() => server.listen());
+afterEach(() => server.resetHandlers());
+afterAll(() => server.close());
+
+it("allows the user to register successfully", async () => {
+  render(
+    <Router>
+      <Register />
+    </Router>
+  );
+
+  fireEvent.change(screen.getByPlaceholderText("Full Name"), {
+    target: { value: "My Name" },
+  });
+  act(() => {
+    fireEvent.change(screen.getByPlaceholderText("Username"), {
```

Comment:
do we really need to use `act()` in this test if none of these form input events are asynchronous except for the submit registration?

Reply:
@rkraeher I guess we need to use act() 
![image](https://user-images.githubusercontent.com/71868375/156923790-7ce72586-bcf6-4d51-af48-3e33f8551ace.png)

Reply:
this warning seems to have to do with state updates happening asynchronously. this is not really how `act` should be used if you check out the [docs](https://reactjs.org/docs/test-utils.html#act). 

[This article](https://davidwcai.medium.com/react-testing-library-and-the-not-wrapped-in-act-errors-491a5629193b) also has a really useful explanation of the error and some suggestions on alternative solutions for how to deal with it effectively and eliminate the warning